### PR TITLE
Python 3 Only

### PR DESCRIPTION
**Pull request by Cadair**
This PR makes a bunch of Python 2 incompatible changes with the objective of closing #2287

### CODE REVIEW

File: docs/guide/installation/mac.rst [hunk 1]
```diff
@@ -24,10 +24,10 @@ install your own version of Python. You will then have two versions of Python li
 your system at the same time. It can be confusing to make sure that when you install
 packages they are installed for the correct Python so beware.
 
-Python 2.7.3+
-^^^^^^^^^^^^^
+Python 3.6+
+^^^^^^^^^^^
 Download and install the latest version of
-`32 bit Python 2.7.3 <https://www.python.org/download/releases/2.7.3/>`
+`32 bit Python 3 <https://www.python.org/download/releases/>`
```
Comment (nabobalis) [inline docs/guide/installation/mac.rst]: 32bit?!

Comment (dpshelio) [inline docs/guide/installation/mac.rst]: and what's the 32 bits for??
```suggestion
`Python 3 <https://www.python.org/download/releases/>`
```

File: docs/guide/installation/mac.rst [hunk 2]
```diff
@@ -24,10 +24,10 @@ install your own version of Python. You will then have two versions of Python li
 your system at the same time. It can be confusing to make sure that when you install
 packages they are installed for the correct Python so beware.
 
-Python 2.7.3+
-^^^^^^^^^^^^^
+Python 3.6+
+^^^^^^^^^^^
 Download and install the latest version of
-`32 bit Python 2.7.3 <https://www.python.org/download/releases/2.7.3/>`
+`32 bit Python 3 <https://www.python.org/download/mac-osx/>`
```
Comment (nabobalis) [inline docs/guide/installation/mac.rst]: Seriously tho, should we not tell them to download the 64 bit one?

Comment (nabobalis) [reply docs/guide/installation/mac.rst]: `macOS 64-bit/32-bit installer`?

Comment (nabobalis) [reply docs/guide/installation/mac.rst]: or `macOS 64-bit installer`

Comment (nabobalis) [reply docs/guide/installation/mac.rst]: The latter is for most up to date systems.

Comment (Cadair) [reply docs/guide/installation/mac.rst]: I don't know, this whole install page is criminally out of date imo.

**Comment by Cadair** [comment]
FTR: This is what the API docs look like with the function annotations: https://5314-2165383-gh.circle-artifacts.com/0/root/project/docs/_build/html/code_ref/map.html#sunpy.map.mapbase.GenericMap.pixel_to_world

**Pull request merged**